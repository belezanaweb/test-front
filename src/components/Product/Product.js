import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

const StyledProduct = styled.div`
  display: flex;
  background-color: #fff;
  border-radius: 3px;
  border: 1px solid #eee;
  padding: 4px 6px 4px 10px;
  margin-bottom: 15px;
`

const StyledProductName = styled.p`
  color: #212122;
  font-size: 13px;
  line-height: 16px;
  word-spacing: 0.9px;
  margin-top: 7px;
  margin-bottom: 17px;
`

const StyledProductImg = styled.img`
  width: 64px;
  height: 100%;
  margin-right: 10px;
  margin-top: 7px;
`

const StyledProductPrice = styled.span`
  margin-right: 3px;
  font-size: 14px;
  font-weight: 700;
  line-height: 17px;
  float: right;
  margin-bottom: 7px;
`

function Product({ product }) {
  const price = new Intl.NumberFormat('pt-BR', { style: 'currency', currency: 'BRL' }).format(
    product.priceSpecification.price
  )
  return (
    <StyledProduct>
      <StyledProductImg src={product.imageObjects[0].small} alt={product.name} />
      <div>
        <StyledProductName>{product.name}</StyledProductName>
        <StyledProductPrice>{price}</StyledProductPrice>
      </div>
    </StyledProduct>
  )
}

Product.propTypes = {
  product: PropTypes.shape({
    sku: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.number,
    imageObjects: PropTypes.array
  }).isRequired
}

export default Product
