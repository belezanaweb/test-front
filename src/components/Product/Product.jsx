import React from 'react'
import { formatToBRL } from '../../utils/currencyHelper'
import * as CSS from './style'

export default function Product({ hidePrice, item }) {
  const { name } = item.product
  const { price } = item.product.priceSpecification
  const { thumbnail, extraLarge } = item.product.imageObjects[0]

  return (
    <>
      <a href={extraLarge} target="_blank" rel="noopener noreferrer">
        <CSS.Img src={thumbnail} alt={name} />
      </a>
      <CSS.Desc>
        <CSS.Label title={name}>{name}</CSS.Label>
        {hidePrice ?? <CSS.Price>{formatToBRL(price)}</CSS.Price>}
      </CSS.Desc>
    </>
  )
}
