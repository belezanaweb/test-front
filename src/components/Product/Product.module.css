.Container,
.PriceContainer {
  display: flex;
  align-items: center;
}

.Container {
  margin-bottom: 40px;
  justify-content: space-around;
}

.PriceContainer {
  flex-direction: column;
}

.ProductName {
  width: 180px;
  line-height: 16px;
}