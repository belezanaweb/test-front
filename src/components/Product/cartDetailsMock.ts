export const CartDetails = [
  {
    product: {
      imageObjects: [
        {
          extraLarge:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/5/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
          featured: true,
          large:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
          medium:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/6/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
          small:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/2/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
          thumbnail:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
          valid: true
        }
      ],
      name:
        "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g",
      priceSpecification: {
        discount: 18,
        maxPrice: 243.9,
        originalPrice: 225.9,
        percent: 7,
        price: 225.9,
        sku: '24410'
      },
      sku: '24410'
    },
    quantity: 1
  },

  {
    product: {
      imageObjects: [
        {
          extraLarge:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/5/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg',
          featured: true,
          large:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg',
          medium:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/6/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg',
          small:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/2/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg',
          thumbnail:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg',
          valid: true
        }
      ],
      name: 'Good Girl Carolina Herrera Eau de Parfum - Perfume Feminino 30ml',
      priceSpecification: {
        discount: 0,
        maxPrice: 299,
        originalPrice: 299,
        percent: 0,
        price: 299,
        sku: '38273'
      },
      sku: '38273'
    },
    quantity: 1
  },

  {
    product: {
      imageObjects: [
        {
          extraLarge:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/5/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png',
          featured: true,
          large:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png',
          medium:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/6/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png',
          small:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/2/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png',
          thumbnail:
            'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png',
          valid: true
        }
      ],
      name: 'Senscience Inner Restore Intensif - Máscara Capilar 50ml',
      priceSpecification: {
        discount: 0,
        maxPrice: 99.9,
        originalPrice: 99.9,
        percent: 0,
        price: 99.9,
        sku: '3019'
      },
      sku: '3019'
    },
    quantity: 1
  }
]
