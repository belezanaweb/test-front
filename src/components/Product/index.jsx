import React from 'react'
import { Container, ImageContainer, Image, TextContainer, ProductTitle, Price } from './styles'

export const Product = ({ name, priceSpecification, imageObjects }) => {
  return (
    <Container isConfirmation={window.location.pathname === '/confirmation'}>
      <ImageContainer>
        <img src={imageObjects[0]?.thumbnail} alt={name} />
      </ImageContainer>
      <TextContainer>
        <ProductTitle>{name}</ProductTitle>
        <Price isConfirmation={window.location.pathname === '/confirmation'}>
          R$ {priceSpecification.price.toFixed(2)}
        </Price>
      </TextContainer>
    </Container>
  )
}
