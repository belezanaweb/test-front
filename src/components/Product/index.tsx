import React, { useMemo } from 'react'
import { toMoney } from '../../utils/money'
import * as S from './styles'

export type ProductProps = {
  image: string,
  description: string,
  price: number
}

export const Product: React.FC<ProductProps> = ({ image, description, price }) => {
  const intoMoney = useMemo(() => {
    return toMoney(price)
  }, [price])

  return (
    <S.Wrapper>
      <S.Figure>
        <img src={image} alt="description" />
      </S.Figure>
      <S.Content>
        <S.Description>
          <p>{description}</p>
        </S.Description>
        <S.Price>
          <span>{intoMoney}</span>
        </S.Price>
      </S.Content>
    </S.Wrapper>
  )
}
