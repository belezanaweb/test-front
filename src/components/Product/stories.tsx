import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import { Product, ProductProps } from '.'

export default {
  title: 'Product',
  component: Product,
  parameters: {
    layout: 'fullscreen'
  }
}

export const Basic: Story<ProductProps> = args => <Product {...args} />

Basic.args = {
  image:
    'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
  description:
    "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g",
  price: 225.9
}
