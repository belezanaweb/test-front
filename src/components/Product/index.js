import Product from './Product';

export default Product;
