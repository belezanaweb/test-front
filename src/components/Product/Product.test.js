import React from 'react'
import { render } from '@testing-library/react'

import data from '../../data/endpoint.json'

import Product from './Product'

describe('Product Component', () => {
  test('renders links', async () => {
    const { asFragment } = render(<Product product={data.items[0].product} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
