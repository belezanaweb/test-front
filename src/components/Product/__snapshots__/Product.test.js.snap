// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component renders links 1`] = `
<DocumentFragment>
  <div
    class="Product__StyledProduct-wun1w2-0 fVLvfH"
  >
    <img
      alt="L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium"
      class="Product__StyledProductImg-wun1w2-2 cetBqs"
      src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/2/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png"
    />
    <div>
      <p
        class="Product__StyledProductName-wun1w2-1 fTfnil"
      >
        L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium
      </p>
      <span
        class="Product__StyledProductPrice-wun1w2-3 dJWezn"
      >
        R$225.90
      </span>
    </div>
  </div>
</DocumentFragment>
`;
